test/core/messages-reactions.integration.test.ts

```diff
@@ -384,4 +384,126 @@ describe('message reactions integration', { timeout: 60000 }, () => {
+    await room2.attach();
+
+    // Second client adds two distinct reactions
+    await room2.messages.reactions.add(message1, { type: MessageReactionType.Distinct, name: '👍' });
+    await room2.messages.reactions.add(message1, { type: MessageReactionType.Distinct, name: '❤️' });
+
+    // Wait for first two summaries (after adding reactions)
+    await vi.waitFor(
+      () => {
+        expect(found.length).toBeGreaterThanOrEqual(2);
+        
+        // Check first summary (after adding 👍)
+        const firstSummary = found.find((e) => 
+          e.summary.messageSerial === message1.serial && 
+          e.summary.distinct?.['👍'] && 
+          !e.summary.distinct?.['❤️']
+        );
+        expect(firstSummary).toMatchObject({
+          type: MessageReactionEvents.Summary,
+          summary: {
+            messageSerial: message1.serial,
+            distinct: {
+              '👍': {
+                total: 1,
+                clientIds: [client2.clientId],
+              },
+            },
+          },
+        });
+
+        // Check second summary (after adding ❤️)
+        const secondSummary = found.find((e) => 
+          e.summary.messageSerial === message1.serial && 
+          e.summary.distinct?.['👍'] && 
+          e.summary.distinct?.['❤️']
+        );
+        expect(secondSummary).toMatchObject({
+          type: MessageReactionEvents.Summary,
+          summary: {
+            messageSerial: message1.serial,
+            distinct: {
+              '👍': {
+                total: 1,
+                clientIds: [client2.clientId],
+              },
+              '❤️': {
+                total: 1,
+                clientIds: [client2.clientId],
+              },
+            },
+          },
+        });
+      },
+      { timeout: 30_000 },
+    );
+
+    // Second client deletes both reactions
+    await room2.messages.reactions.delete(message1, { type: MessageReactionType.Distinct, name: '👍' });
+    await room2.messages.reactions.delete(message1, { type: MessageReactionType.Distinct, name: '❤️' });
+
+    // Wait for all 4 summaries
+    await vi.waitFor(
+      () => {
+        expect(found.length).toBeGreaterThanOrEqual(4);
+        
+        // Check third summary (after deleting 👍)
+        const thirdSummary = found.find((e) => 
+          e.summary.messageSerial === message1.serial && 
+          !e.summary.distinct?.['👍'] && 
+          e.summary.distinct?.['❤️']
+        );
+        expect(thirdSummary).toMatchObject({
+          type: MessageReactionEvents.Summary,
+          summary: {
+            messageSerial: message1.serial,
+            distinct: {
+              '❤️': {
+                total: 1,
+                clientIds: [client2.clientId],
+              },
+            },
+          },
+        });
+
+        // Check fourth summary (after deleting ❤️ - should be empty or not have these reactions)
+        const fourthSummary = found.findLast((e) => e.summary.messageSerial === message1.serial);
```

Comment:
If the third one is not at index 2 (third in list), you'll find the third one here.

Would it be better to just findIndex the third one, and get the other index for the fourth one?

We're talking about indexes 2 and 3 for those checks, or 0 and 1 if you clear the found array.

I guess the same logic for finding the summaries in the list can be done for the first two as well?